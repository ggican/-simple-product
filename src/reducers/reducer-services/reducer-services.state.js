const serviceState = {
    recipes: {
        ingredients: false,
    },
    user: {
        auth: false,
    },
    product: {
        productList: false,
        purchase: [],
    },
};

export default serviceState;
