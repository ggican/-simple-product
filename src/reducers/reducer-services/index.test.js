import React from "react";
import { servicesRedux, servicesState } from "./reducer-services.redux";

describe("Reducer Services Component", () => {
    const state = servicesState;
    beforeEach(() => {});

    it("SUCCESS servicesRedux type 200 function and Snapshot servicesRedux", () => {
        const actionTest = {
            reducer: "reducer",
            group: "recipes",
            key: "ingredients",
            data: ["cool"],
            message: {
                message: "cool its me",
            },
            type: 200,
            isError: false,
            isSuccess: true,
        };

        expect(servicesRedux(servicesState, actionTest)).toMatchSnapshot();
    });
    it("SUCCESS servicesState  Snapshot servicesState", () => {
        expect(servicesState).toMatchSnapshot();
    });
    it("SUCCESS servicesRedux type 200 function servicesRedux", () => {
        const actionTest = {
            reducer: "reducer",
            group: "recipes",
            key: "ingredients",
            data: ["cool"],
            message: {
                message: "cool its me",
            },
            type: 200,
            isError: false,
            isSuccess: true,
        };
        expect(servicesRedux(state, actionTest)).toEqual({
            recipes: {
                ingredients: actionTest,
            },
        });
    });

    it("SUCCESS servicesRedux type 201 function servicesRedux", () => {
        const actionTest = {
            reducer: "reducer",
            group: "recipes",
            key: "ingredients",
            data: ["cool"],
            message: {
                message: "cool its me",
            },
            type: 201,
            isError: false,
            isSuccess: true,
        };
        expect(servicesRedux(state, actionTest)).toEqual({
            recipes: {
                ingredients: actionTest,
            },
        });
    });

    it("SUCCESS servicesRedux type 201 function servicesRedux", () => {
        const actionTest = {
            reducer: "reducer",
            group: "recipes",
            key: "ingredients",
            data: ["cool"],
            message: {
                message: "cool its me",
            },
            type: 204,
            isError: false,
            isSuccess: true,
        };
        expect(servicesRedux(state, actionTest)).toEqual({
            recipes: {
                ingredients: actionTest,
            },
        });
    });

    it("SUCCESS servicesRedux type 201 function servicesRedux", () => {
        const actionTest = {
            reducer: "reducer",
            group: "recipes",
            key: "ingredients",
            data: ["cool"],
            message: {
                message: "cool its me",
            },
            type: 400,
            isError: true,
            isSuccess: false,
        };
        expect(servicesRedux(state, actionTest)).toEqual({
            recipes: {
                ingredients: actionTest,
            },
        });
    });

    it("SUCCESS servicesRedux type 401 function servicesRedux", () => {
        const actionTest = {
            reducer: "reducer",
            group: "recipes",
            key: "ingredients",
            data: ["cool"],
            message: {
                message: "cool its me",
            },
            type: 401,
            isError: true,
            isSuccess: false,
        };
        expect(servicesRedux(state, actionTest)).toEqual({
            recipes: {
                ingredients: actionTest,
            },
        });
    });

    it("SUCCESS servicesRedux type 404 function servicesRedux", () => {
        const actionTest = {
            reducer: "reducer",
            group: "recipes",
            key: "ingredients",
            data: ["cool"],
            message: {
                message: "cool its me",
            },
            type: 404,
            isError: true,
            isSuccess: false,
        };
        expect(servicesRedux(state, actionTest)).toEqual({
            recipes: {
                ingredients: actionTest,
            },
        });
    });

    it("SUCCESS servicesRedux type 409 function servicesRedux", () => {
        const actionTest = {
            reducer: "reducer",
            group: "recipes",
            key: "ingredients",
            data: ["cool"],
            message: {
                message: "cool its me",
            },
            type: 409,
            isError: true,
            isSuccess: false,
        };
        expect(servicesRedux(state, actionTest)).toEqual({
            recipes: {
                ingredients: actionTest,
            },
        });
    });

    it("SUCCESS servicesRedux type 500 function servicesRedux", () => {
        const actionTest = {
            reducer: "reducer",
            group: "recipes",
            key: "ingredients",
            data: ["cool"],
            message: {
                message: "cool its me",
            },
            type: 500,
            isError: true,
            isSuccess: false,
        };
        expect(servicesRedux(state, actionTest)).toEqual({
            recipes: {
                ingredients: actionTest,
            },
        });
    });
    it("SUCCESS servicesRedux type CLEAR function servicesRedux", () => {
        const actionTest = {
            reducer: "reducer",
            group: "recipes",
            key: "ingredients",
            defaultData: {
                isDefault: true,
            },
            type: "CLEAR",
        };
        expect(servicesRedux(state, actionTest)).toEqual({
            recipes: { ingredients: { isDefault: true } },
        });
    });
    it("SUCCESS servicesRedux type DEFAULT function servicesRedux", () => {
        const actionTest = {
            type: false,
        };
        expect(servicesRedux(state, actionTest)).toEqual({
            recipes: { ingredients: false },
        });
    });
});
