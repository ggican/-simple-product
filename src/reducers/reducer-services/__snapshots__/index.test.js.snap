// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reducer Services Component SUCCESS servicesRedux type 200 function and Snapshot servicesRedux 1`] = `
Object {
  "recipes": Object {
    "ingredients": Object {
      "data": Array [
        "cool",
      ],
      "group": "recipes",
      "isError": false,
      "isSuccess": true,
      "key": "ingredients",
      "message": Object {
        "message": "cool its me",
      },
      "reducer": "reducer",
      "type": 200,
    },
  },
}
`;

exports[`Reducer Services Component SUCCESS servicesState  Snapshot servicesState 1`] = `
Object {
  "recipes": Object {
    "ingredients": false,
  },
}
`;
